src/hdmf/common/io/table.py

```diff
@@ -82,6 +82,8 @@ def process_field_spec(cls, classdict, docval_args, parent_cls, attr_name, not_i
             # the spec does not know which table this DTR points to
             # the user must specify the table attribute on the DTR after it is generated
             column_conf['table'] = True
+        else:
+            column_conf['class'] = dtype
```

Comment:
I *think* this flow would work for subclasses of `VectorData` but it would be good to test that.

Reply:
Do you mean for stuff like `EnumData`, `DynamicTableRegion`, etc?

Reply:
yeah, well not `DynamicTableRegion`, because that's handled above, and I am not sure `EnumData` is fully handled by `__columns__` but I think it is. I was more thinking of custom extensions to VectorData. Making such a test might be a pain though. I think it would be fine but feel free to add a test.